harmonica/tests/test_icgem.py

```diff
@@ -0,0 +1,286 @@
+import numpy.testing as npt
+from pytest import raises
+
+from .. import load_icgem_gdf
+
+MODULE_DIR = os.path.dirname(__file__)
+TEST_DATA_DIR = os.path.join(MODULE_DIR, "data")
+
+
+def test_load_icgem_gdf():
+    "Check if load_icgem_gdf reads an ICGEM file with sample data correctly"
+    fname = os.path.join(TEST_DATA_DIR, "icgem-sample.gdf")
+    icgem_grd = load_icgem_gdf(fname)
+
+    s, n, w, e = 16, 28, 150, 164
+    nlat, nlon = 7, 8
+    shape = (nlat, nlon)
+    lat = np.linspace(s, n, nlat, dtype="float64")
+    lon = np.linspace(w, e, nlon, dtype="float64")
+    lon, lat = np.meshgrid(lon, lat)
+    true_data = np.array([np.arange(nlon)] * nlat, dtype="float64")
+    height = 1100 * np.ones(shape)
+
+    assert icgem_grd.dims["northing"] == nlat
+    assert icgem_grd.dims["easting"] == nlon
+    npt.assert_equal(icgem_grd.lon.values, lon)
+    npt.assert_equal(icgem_grd.lat.values, lat)
+    npt.assert_allclose(true_data, icgem_grd.sample_data.values)
+    npt.assert_allclose(height, icgem_grd.height.values)
+
+
+def test_load_icgem_gdf_with_height():
+    "Check if load_icgem_gdf reads an ICGEM file with height column"
+    fname = os.path.join(TEST_DATA_DIR, "icgem-sample-with-height.gdf")
+    icgem_grd = load_icgem_gdf(fname)
+
+    s, n, w, e = 16, 28, 150, 164
+    nlat, nlon = 7, 8
+    lat = np.linspace(s, n, nlat, dtype="float64")
+    lon = np.linspace(w, e, nlon, dtype="float64")
+    lon, lat = np.meshgrid(lon, lat)
+    true_data = np.array([np.arange(nlon)] * nlat, dtype="float64")
+    height = lon + lat
+
+    assert icgem_grd.dims["northing"] == nlat
+    assert icgem_grd.dims["easting"] == nlon
+    npt.assert_equal(icgem_grd.lon.values, lon)
+    npt.assert_equal(icgem_grd.lat.values, lat)
+    npt.assert_allclose(true_data, icgem_grd.sample_data.values)
+    npt.assert_allclose(height, icgem_grd.h_over_geoid.values)
+
+
+def test_load_icgem_gdf_usecols():
+    "Check if load_icgem_gdf reads an ICGEM file reading only first two columns"
+    fname = os.path.join(TEST_DATA_DIR, "icgem-sample.gdf")
+    icgem_grd = load_icgem_gdf(fname, usecols=[0, 1])
+
+    s, n, w, e = 16, 28, 150, 164
+    nlat, nlon = 7, 8
+    shape = (nlat, nlon)
+    lat = np.linspace(s, n, nlat, dtype="float64")
+    lon = np.linspace(w, e, nlon, dtype="float64")
+    lon, lat = np.meshgrid(lon, lat)
+    height = 1100 * np.ones(shape)
+
+    assert icgem_grd.dims["northing"] == nlat
+    assert icgem_grd.dims["easting"] == nlon
+    npt.assert_equal(icgem_grd.lon.values, lon)
+    npt.assert_equal(icgem_grd.lat.values, lat)
+    npt.assert_allclose(height, icgem_grd.height.values)
+    assert len(icgem_grd.data_vars) == 1
+
+
+def test_missing_shape(tmpdir):
+    "ICGEM file with missing shape"
+    fname = os.path.join(TEST_DATA_DIR, "icgem-sample.gdf")
+    attributes = ["latitude_parallels", "longitude_parallels"]
+    for attribute in attributes:
+        corrupt = tmpdir.join("missing_shape_" + attribute + ".gdf")
```

Comment:
```suggestion
        corrupt = str(tmpdir.join("missing_shape_" + attribute + ".gdf"))
```

Turning the `tmpdir` into a string should be enough to satisfy Python 3.5. I think that `open` can only take non-string arguments starting in 3.6.